packages/aws-cdk/lib/cli.ts

```diff
@@ -367,9 +367,11 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
   });
   await configuration.load();
 
-  if (shouldDisplayNotices()) {
-    void refreshNotices()
-      .catch(e => debug(`Could not refresh notices: ${e}`));
+  const cmd = argv._[0];
+
+  const notices = Notices.create({ configuration, includeAcknowlegded: cmd === 'notices' ? !argv.unacknowledged : false });
+  if (notices.shouldDisplay()) {
+    void notices.refresh().catch(e => debug(`Could not refresh notices: ${e}`));
```

Comment:
Similar point about encapsulation and non-intrusiveness. `refresh` should never fail on the caller. Any failures should be caught and debugged internally.

Reply:
Nice one.